### DISCUSSION TITLE
Custom css wont be applied

### DISCUSSION THREAD

**Discussion by avifatal** [rsuite/rsuite#2048] [General]
Hi,
I have this in my index.tsx:
```
import { Button } from 'rsuite';
import 'rsuite/dist/rsuite.min.css'
import './index.less';
```
I have this in my index.less:
```
@H050:#fff5f2;
@H100:#facfc3;
@H200:#f5a890;
@H300:#f08460;
@H400:#eb6331;
@H500:#e54304;
@H600:#db4302;
@H700:#c73f00;
@H800:#a83800;
@H900:#802d00;
```
BTW, If I import `rsuite/styles/index.less` I see plain HTML button.

How Can I apply my custom coloring?

Thanks

**Comment by SevenOutman** [comment]
You should import RSuite Less styles if you need to customize styles by overriding Less variables.

> BTW, If I import rsuite/styles/index.less I see plain HTML button.

Did you setup webpack with tools like `ExtractTextPlugin` to extract styles into standalone stylesheet files?

Here's an example configuration https://rsuitejs.com/guide/customization/#webpack-compiles-multiple-themes

**Reply by avifatal** [reply]
Cant I provide these colors via javascript in runtime? thanks

**Reply by avifatal** [reply]
@SevenOutman , just mentioning that I want it to be dynamic, so maybe passing it as js parama would be better.
Thanks

**Reply by SevenOutman** [reply]
> Cant I provide these colors via javascript in runtime? thanks

It's not easy to modify Less variables at runtime (you need to load Less files and run Less compiler in your browser). So if you need multiple sets of colors using Less variables, I'd recommend pre-compile them into separate CSS files, and load them as demanded.

One way to switch color themes at runtime is using CSS custom properties. Most color styles are assigned via CSS custom properties in RSuite, so you can change values of the custom properties to tweak how RSuite looks.

**Reply by SevenOutman** [reply]
> Thank you. Do you have an example? this concept is kind of new for me... or maybe I got you wrong

You can refer to [CSS Custom Properties on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) to learn the basic usage.

RSuite defines a set of custom properties under the `.rs-theme-light` className, you may find a full list of them in your Chrome Devtools by inspecting an element rendered by RSuite components.
![image](https://user-images.githubusercontent.com/8225666/138845483-893dc6ee-aa3a-4719-b47f-e056acaa8ca9.png)

You can override these properties by re-defining them in a custom className and apply that class to `<body>`.
